src/NLog/LogFactoryExtensions.cs

```diff
@@ -0,0 +1,133 @@
+﻿// 
+// Copyright (c) 2004-2019 Jaroslaw Kowalski <jaak@jkowalski.net>, Kim Christensen, Julian Verdurmen
+// 
+// All rights reserved.
+// 
+// Redistribution and use in source and binary forms, with or without 
+// modification, are permitted provided that the following conditions 
+// are met:
+// 
+// * Redistributions of source code must retain the above copyright notice, 
+//   this list of conditions and the following disclaimer. 
+// 
+// * Redistributions in binary form must reproduce the above copyright notice,
+//   this list of conditions and the following disclaimer in the documentation
+//   and/or other materials provided with the distribution. 
+// 
+// * Neither the name of Jaroslaw Kowalski nor the names of its 
+//   contributors may be used to endorse or promote products derived from this
+//   software without specific prior written permission. 
+// 
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE 
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE 
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE 
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR 
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS 
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN 
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) 
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF 
+// THE POSSIBILITY OF SUCH DAMAGE.
+// 
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using NLog.Common;
+using NLog.Config;
+using NLog.Internal;
+
+namespace NLog
+{
+    /// <summary>
+    /// Extensions for NLog <see cref="LogFactory"/>.
+    /// </summary>
+    public static class LogFactoryExtensions
+    {
+        /// <summary>
+        /// Registers object Type transformation from dangerous (massive) object to safe (reduced) object
+        /// </summary>
+        public static LogFactory RegisterObjectTransformation<T>(this LogFactory logFactory, Func<T, object> transformer)
```

Comment:
Not sure if the names are really clear or even to long, we could also do something like this:

`LogFactory.SerializationOptions.RegisterTransformation(...)`

(and also maybe `LogFactory.SerializationOptions.SkipSerialization(type)`)

Reply:
it's nice grouped then. If we need it on LogFactory, NLog or maybe LogManager is also something we could think of

Reply:
These a helper methods to hide away the dependency-injection chains. Suddenly making a bucket for certain interfaces called SerializationOptions will not be good.

Reply:
Don't you think LogFactory gets to many methods which aren't related to each other?

Reply:
I hope LogFactory will have all the methods for configuring NLog and for creating Logger-objects.

Reply:
I have now added a new commit with `IConfigBuilder`. Then all extension methods related to updating LogFactory-configuration can be made as fluent extension methods for this interface. Is that an improvement?

Reply:
Renamed from `Config()` to `Setup()` to avoid confusion with `LogManager.Configuration`